### DISCUSSION TITLE
Help with direct control from Pi of WS2812s

### DISCUSSION THREAD

**Discussion by FuzzyMistborn** [hyperion-project/hyperion.ng#1362] [Q&A] [answered]
I feel like I must be missing something obvious here.  Currently I run Hyperion on a Pi and then use WLED to directly drive the LEDs themselves (WS2812B's).  It works mostly OK but the wifi lag on the ESP/WLED is annoying and I get occasional disconnects.  Thinking of wiring the Pi in directly to the LEDs.

My current wiring is something like this:
![image](https://user-images.githubusercontent.com/16689090/138568555-267a8e7d-02b4-424c-90a8-10f55ca27fcc.png)

The main difference is I power the ESP board over the USB, so no 5v/ground from the power supply to the ESP board.  Basically 3v, ground, and a data wire go from the ESP through a logic level shifter (low side for the ESP, high to the power supply) to the LEDs.  The issue is when I unplug from the ESP and move the wires to the Pi (same thing, 3v, ground, and to GPIO 18) I get nothing.  Can't even do a manual effect/control the LEDs that way.  I've change the setting in Hyperion to "ws281x" for LED controller type, clicked save...nothing.  Do I need to set something on the Pi for it to work?  Install something?

**Comment by Lord-Grey** [comment, marked as answer]
@FuzzyMistborn I suggest you post your question in the [Forum](https://hyperion-project.org/forum/)  too.
There are many knowledgeable people which have similar setups and can advise accordingly.

**Reply by FuzzyMistborn** [reply]
Done, thank you

https://hyperion-project.org/forum/index.php?thread/12328-help-with-direct-control-from-pi-of-ws2812s/

**Reply by FuzzyMistborn** [reply]
Solved.  https://hyperion-project.org/forum/index.php?thread/12328-help-with-direct-control-from-pi-of-ws2812s/&postID=42837#post42837
